overlord/assertstate/assertmgr.go

```diff
@@ -189,10 +202,79 @@ func doFetchSnapAssertions(t *state.Task, _ *tomb.Tomb) error {
 		PrimaryKey: []string{sha3_384},
 	}
 
-	return fetch(t.State(), ref, ss.UserID)
+	err = fetch(t.State(), ref, ss.UserID)
+	if notFound, ok := err.(*assertionNotFoundError); ok {
+		if notFound.ref.Type == asserts.SnapRevisionType {
+			return fmt.Errorf("cannot verify snap %q and its hash, no matching assertions found", ss.Name())
```

Comment:
I suggest s/and its hash// here and below. The hash _is_ the snap.

Also suggest replacing the word "assertions" here and below by "signatures". Although a bit less precise for us, it's not completely wrong and it's a much more common term in this context.